source/opaque.d.ts

```diff
@@ -113,19 +113,24 @@ type WillWork = UnwrapOpaque<Tagged<number, 'AccountNumber'>>; // number
 @category Type
 */
 export type UnwrapOpaque<OpaqueType extends TagContainer<unknown>> =
-	OpaqueType extends MultiTagContainer<string | number | symbol>
-		? RemoveAllTags<OpaqueType>
-		: OpaqueType extends Opaque<infer Type, OpaqueType[typeof tag]>
-			? Type
-			: OpaqueType;
+	OpaqueType extends Opaque<infer Type, OpaqueType[typeof tag]>
+		? Type
+		: OpaqueType;
```

Comment:
FYI, this updated implementation of `UnwrapOpaque` still works on `Tagged` types, and there's a test for that. The new implementation is just simpler, by exploiting the fact that `TagContainer` and `MultiTagContainer` share the same basic structure. It lets us remove `RemoveAllTags`, which was causing some circular type errors when used with tag metadata.